variod.c

```diff
@@ -84,7 +84,7 @@ void INThandler(int sig)
 
 static void wait_for_XCSoar(int xcsoar_sock, sockaddr* s_xcsoar)
 {
-	while (connect(xcsoar_sock, s_xcsoar, sizeof(*s_xcsoar)) < 0) {
+	while (connect(xcsoar_sock, s_xcsoar, sizeof(struct sockaddr_in)) < 0) {
```

Comment:
I don't understand this part. You passing different size than reserved for `s_xcsoar` - this doesn't look right. Why?